graft/coreth/plugin/evm/vmsync/client.go

```diff
@@ -356,52 +305,97 @@ func (client *client) finishSync() error {
+	c.config.Chain.BloomIndexer().AddCheckpoint(parentHeight/params.BloomBitsBlocks, parentHash)
 
-	if err := client.Chain.BlockChain().ResetToStateSyncedBlock(block); err != nil {
+	if err := ctx.Err(); err != nil {
+		return fmt.Errorf("%w: %w", errCommitCancelled, err)
+	}
+	if err := c.config.Chain.BlockChain().ResetToStateSyncedBlock(block); err != nil {
 		return err
 	}
 
-	if client.Extender != nil {
-		if err := client.Extender.OnFinishBeforeCommit(client.LastAcceptedHeight, client.summary); err != nil {
+	if c.config.Extender != nil {
+		if err := c.config.Extender.OnFinishBeforeCommit(c.config.LastAcceptedHeight, summary); err != nil {
 			return err
 		}
 	}
 
-	if err := client.commitVMMarkers(); err != nil {
-		return fmt.Errorf("error updating vm markers, height=%d, hash=%s, err=%w", block.NumberU64(), block.Hash(), err)
+	if err := c.commitMarkers(summary); err != nil {
+		return fmt.Errorf("%w: height=%d, hash=%s: %w", errCommitMarkers, block.NumberU64(), block.Hash(), err)
 	}
 
-	if err := client.State.SetLastAcceptedBlock(wrappedBlock); err != nil {
+	if err := c.config.State.SetLastAcceptedBlock(wrappedBlock); err != nil {
 		return err
 	}
 
-	if client.Extender != nil {
-		return client.Extender.OnFinishAfterCommit(block.NumberU64())
+	if c.config.Extender != nil {
+		if err := c.config.Extender.OnFinishAfterCommit(block.NumberU64()); err != nil {
+			return err
+		}
 	}
 
 	return nil
 }
 
-// commitVMMarkers updates the following markers in the VM's database
-// and commits them atomically:
-// - updates atomic trie so it will have necessary metadata for the last committed root
-// - updates atomic trie so it will resume applying operations to shared memory on initialize
-// - updates lastAcceptedKey
-// - removes state sync progress markers
-func (client *client) commitVMMarkers() error {
-	// Mark the previously last accepted block for the shared memory cursor, so that we will execute shared
-	// memory operations from the previously last accepted block to [vm.syncSummary] when ApplyToSharedMemory
-	// is called.
-	id := ids.ID(client.summary.GetBlockHash())
-	if err := client.Acceptor.PutLastAcceptedID(id); err != nil {
+// commitMarkers updates VM database markers atomically.
+func (c *client) commitMarkers(summary message.Syncable) error {
+	id := ids.ID(summary.GetBlockHash())
+	if err := c.config.Acceptor.PutLastAcceptedID(id); err != nil {
 		return err
 	}
-	if err := client.MetadataDB.Delete(stateSyncSummaryKey); err != nil {
+	if err := c.config.MetadataDB.Delete(stateSyncSummaryKey); err != nil {
 		return err
 	}
-	return client.VerDB.Commit()
+	return c.config.VerDB.Commit()
 }
 
-// Error returns a non-nil error if one occurred during the sync.
-func (client *client) Error() error { return client.err }
+// newSyncerRegistry creates a registry with all required syncers for the given summary.
+func (c *client) newSyncerRegistry(summary message.Syncable) (*SyncerRegistry, error) {
```

Comment:
(no action required): there are bunch of code moving around. the diff could have been kept less than what it shows. it kind-of hinders the actual meaningful changes (and actually could hide potentially invalid changes as well)